e2e-tests-rust/tests/lib.rs

```diff
@@ -32,45 +33,28 @@ async fn init(
     Ok(provider)
 }
 
-#[tokio::test]
-async fn interval_sealing_finalization() -> anyhow::Result<()> {
-    // Test that we can submit a transaction and wait for it to finalize when era-test-node is
-    // operating in interval sealing mode.
-    let provider = init(|node| node.block_time(1)).await?;
-
-    let tx = TransactionRequest::default()
-        .with_to(address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045"))
-        .with_value(U256::from(100));
-    let receipt = provider.send_transaction(tx).await?.get_receipt().await?;
-    assert!(receipt.status());
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn interval_sealing_multiple_txs() -> anyhow::Result<()> {
-    // Test that we can submit two transactions and wait for them to finalize in the same block when
-    // era-test-node is operating in interval sealing mode. 3 seconds should be long enough for
-    // the entire flow to execute before the first block is produced.
-    let provider = init(|node| node.block_time(3)).await?;
-    const RICH_WALLET0: Address = address!("f39Fd6e51aad88F6F4ce6aB8827279cffFb92266");
-    const RICH_WALLET1: Address = address!("70997970C51812dc3A010C7d01b50e0d17dc79C8");
-    const TARGET: Address = address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045");
+const RICH_WALLET0: Address = address!("f39Fd6e51aad88F6F4ce6aB8827279cffFb92266");
+const RICH_WALLET1: Address = address!("70997970C51812dc3A010C7d01b50e0d17dc79C8");
 
+async fn test_finalize_two_txs_in_the_same_block(
+    provider: impl Provider<Http<reqwest::Client>, Zksync> + WalletProvider<Zksync> + Clone + 'static,
+    target: Address,
+) -> anyhow::Result<()> {
     async fn submit_tx(
         provider: impl Provider<Http<reqwest::Client>, Zksync> + WalletProvider<Zksync> + Clone,
         rich_wallet: Address,
+        target: Address,
     ) -> Result<PendingTransaction, PendingTransactionError> {
         let tx = TransactionRequest::default()
             .with_from(rich_wallet)
-            .with_to(TARGET)
+            .with_to(target)
             .with_value(U256::from(100));
         provider.send_transaction(tx).await?.register().await
     }
 
     // Submit two txs at the same time
-    let handle0 = tokio::spawn(submit_tx(provider.clone(), RICH_WALLET0));
-    let handle1 = tokio::spawn(submit_tx(provider.clone(), RICH_WALLET1));
+    let handle0 = tokio::spawn(submit_tx(provider.clone(), RICH_WALLET0, target));
+    let handle1 = tokio::spawn(submit_tx(provider.clone(), RICH_WALLET1, target));
```

Comment:
Rather then hardcoding RICH_WALLET usage do we want to:

```
let accounts = provider.get_accounts().await?;
let account0 = accounts[0];
let account1 = accounts[1];
```

And use those instead?

Reply:
Good point and I tried to do this but unfortunately there is a discrepancy between which private keys `alloy-zksync` is aware of and what `eth_accounts` returns. The reason being legacy rich wallets: `alloy-zksync` relies on CLI output to grab private keys which legacy rich wallets are not a part of while `era-test-node` returns both new and legacy rich wallets from `eth_accounts` (the order in undeterministic so they are intertwined with one another).